### PR TITLE
Makefile.unix, binfmt: add support for cpp with common binary

### PR DESCRIPTION
**Pull request by abhishek-samsung**
Include LIBSUPCXX and cpp library to common binary and add support for static constructors in common binary

```
binary_manager_load: common Header Checking Success
binary_manager_load_binary: Load success! [Name: common] [Version: 200204] [Partition: A] [Compressed Binary]
binary_manager_load: app1 Header Checking Success
binary_manager_load_binary: Load success! [Name: app1] [Version: 190412] [Partition: A] [Compressed Binary]
binary_manager_load: app2 Header Checking Success
System Information:
        Version:
                Platform: 3.1   Binary: 200204
        Commit Hash: 2e5601c
        Build User: root@=
        Build Time: 2023-06-08 14:51:07 IST
        System Time: 06 Dec 2011, 00:00:01 [s] UTC
This is WIFI App

Select Test Scenario.
        -Press R or r : Recovery Test
        -Press U or u : Binary Update Test
        -Press X or x : Terminate Tests.
TASH>>binary_manager_load_binary: Load success! [Name: app2] [Version: 190412] [Partition: A] [Compressed Binary]
Test will be finished.

TASH>>helloxx
TASH>>helloxx_main: Saying hello from the dynamically constructed instance
CHelloWorld::HelloWorld: Hello, World!!
helloxx_main: Saying hello from the instance constructed on the stack
CHelloWorld::HelloWorld: Hello, World!!
helloxx_main: Saying hello from the statically constructed instance
CHelloWorld::HelloWorld: Hello, World!!
```

### CODE REVIEW

**Review by sunghan-chang — APPROVED** [review]

**Comment by sunghan-chang** [comment]
@kishore-sn Could you review this?

**Comment by sunghan-chang** [comment]
@sangwon03 @Taejun-Kwon Please check this for c++ on the loadable.

File: build/configs/rtl8721csm/cpp_loadable/defconfig [hunk 1]
```diff
@@ -24,7 +24,9 @@ CONFIG_TOOLS_DIR="../tools"
 CONFIG_APP_BINARY_SEPARATION=y
 CONFIG_NUM_APPS=5
 # CONFIG_APPS_RAM_REGION_SHAREABLE is not set
-# CONFIG_SUPPORT_COMMON_BINARY is not set
+CONFIG_SUPPORT_COMMON_BINARY=y
```
Comment (kishore-sn) [inline build/configs/rtl8721csm/cpp_loadable/defconfig:27]: Do we need to support common binary by default?

File: os/binfmt/binfmt_execmodule.c [hunk 2]
```diff
@@ -128,6 +128,20 @@ static void exec_ctors(FAR void *arg)
...
+		ctor = g_lib_binp->ctors;
+		for (i = 0; i < g_lib_binp->nctors; i++) {
+			binfo("Calling ctor %d at %p\n", i, (FAR void *)ctor);
+
+			(*ctor)();
+			ctor++;
+		}
+		g_lib_binp->run_library_ctors = false;
+	}
+#endif
```
Comment (kishore-sn) [inline os/binfmt/binfmt_execmodule.c:143]: Would there be a particular order in which ctors need to be called?
I think if common binary is present, then classes and variables defined in common binary would be used by the code in app binary. In that case, would it be better to call ctors of common binary first, and the app binary later on?

Comment (abhishek-samsung) [reply os/binfmt/binfmt_execmodule.c:136]: Done.

File: build/configs/rtl8721csm/cpp_loadable/defconfig (hunk 1 above)
Comment (abhishek-samsung) [reply build/configs/rtl8721csm/cpp_loadable/defconfig:27]: I think it's good to set common binary config as we are using loadable with common binary everywhere else. So, we can test cpp-related things easily on loadable.

**Review by sangwon03 — APPROVED** [review]
LGTM

**Pull request merged**